DOC improve documentation of RandomUnderSampler (#1019)
scikit-learn-contrib/imbalanced-learn: doc/under_sampling.rst

Controlled under-sampling techniques
------------------------------------

> Controlled under-sampling techniques reduce the number of observations from the majority

Reviewer: I prefer to use directly targeted classes without the term "majority"

  >>> print(np.vstack(np.unique([tuple(row) for row in X_resampled], axis=0)).shape)
  (181, 2)

> :class:`RandomUnderSampler` can undersample numerical and also categorical variables
> (i.e., where the values are strings)::

Reviewer: I prefer a middle ground:

```suggestion
:class:`RandomUnderSampler` handles heterogeneous data types, i.e. numerical, categorical, date, etc.::
```

We need to have the same notes for ROS.

  >>> print(y_resampled)
  [0 1]

> :class:`RandomUnderSampler` can also take a pandas dataframe as input for

Reviewer: Actually, all samplers support it. It might be better to just remove this statement and make sure that we have a section about it.

```suggestion
:class:`RandomUnderSampler` also supports pandas dataframes as input for
```
